locust/test/test_dispatch.py

```diff
@@ -3273,6 +3273,29 @@ class User5(User):
             user_classes=[User1, User2, User3, User4, User5],
         )
 
+    def test_remove_missing_workers_with_dead_heartbeat(self):
+        class User1(User):
+            weight = 1
+
+        class User2(User):
+            weight = 1
+
+        class User3(User):
+            weight = 1
+
+        user_classes = [User1, User2, User3]
+
+        worker_nodes = [WorkerNode(str(i + 1)) for i in range(3)]
+        worker_nodes[1].heartbeat = HEARTBEAT_DEAD - 1
+
+        users_dispatcher = UsersDispatcher(worker_nodes=worker_nodes, user_classes=user_classes)
+
+        users_dispatcher.new_dispatch(target_user_count=9, spawn_rate=3)
+        users_dispatcher._wait_between_dispatch = 0
+        users_dispatcher.prepare_rebalance()
+        list(users_dispatcher)
+        self.assertIs(len(users_dispatcher._worker_nodes), 2, "Expected worker 2 will be removed from worker nodes")
```

Comment:
Can you also validate that the remaining workers get the right number of users?

Reply:
added the test case